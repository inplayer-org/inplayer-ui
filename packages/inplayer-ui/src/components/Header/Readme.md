The Header represents a versatile component which can house a title or a TabNavigation, as well as a UserMenu and Additional User Menu.

```jsx static
<Header
  title="My app title"
  userMenuProps={{
    menuTitle: 'User name',
    image:
      'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
    menuItems: [
      { title: 'Hello', onClick: console.log },
      { title: 'World', href: 'http://www.example.com' },
    ],
    actionItem: { title: 'Logout' },
  }}
  additionalUserMenuProps={{
    menuTitle: 'Added menu',
    menuItems: [
      { title: 'Agent1', onClick: console.log },
      { title: 'Agent2',  href: 'http://www.example.com' },
    ],
    actionItem: {
      title: 'See more on this link',
      href: 'http://www.example.com',
      smallSize: true
    },
  }}
  tabNavigationProps={{
    tabs: [
      {
        title: 'Hello',
      },
      {
        title: 'Tabs',
      },
    ],
    selectedTabIndex: 0,
    onTabClick: console.log,
  }}
/>
```

### With all 4 components

```jsx
<Header
  title="My app title"
  userMenuProps={{
    menuTitle: 'User name',
    image:
      'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
    menuItems: [
      { title: 'Hello', onClick: console.log },
      { title: 'World', href: 'http://www.example.com' },
    ],
    actionItem: { title: 'Logout' },
  }}
  additionalUserMenuProps={{
    menuTitle: 'Added menu',
    menuItems: [
      {
        image: 'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
        title: 'Agent1',
        onClick: console.log
      },
      {
        image:
        'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
        title: 'Agent2',
        href: 'http://www.example.com'
      },
    ],
    actionItem: {
        title: 'See more on this link',
        href: 'http://www.example.com',
        smallSize: true
      },
  }}
  tabNavigationProps={{
    tabs: [
      {
        title: 'Hello',
      },
      {
        title: 'Tabs',
      },
    ],
    selectedTabIndex: 0,
    onTabClick: console.log,
  }}
/>
```

### With title and user menu

```jsx
<Header
  title="My app title"
  userMenuProps={{
    image:
      'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
    menuItems: [
      { title: 'Hello', onClick: console.log },
      { title: 'World', href: 'http://www.example.com' },
    ],
    actionItem: { title: 'Logout' },
  }}
/>
```

### With tab navigation and user menu

```jsx
<Header
  userMenuProps={{
    image:
      'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
    menuItems: [
      { title: 'Hello', onClick: console.log },
      { title: 'World', href: 'http://www.example.com' },
    ],
    actionItem: { title: 'Logout' },
  }}

  tabNavigationProps={{
    tabs: [
      {
        title: 'Hello',
      },
      {
        title: 'Tabs',
      },
    ],
    selectedTabIndex: 0,
    onTabClick: console.log,
  }}
/>
```
