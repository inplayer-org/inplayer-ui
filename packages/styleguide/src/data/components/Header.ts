import { ComponentData } from "../types";

const usage = `
  <Header
    title="My app title"
    userMenuProps={{
      menuTitle: 'User name',
      image:
        'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
      menuItems: [
        { title: 'Hello', onClick: console.log },
        { title: 'World', href: 'http://www.example.com' },
      ],
      actionItem: { title: 'Logout' },
    }}
    additionalUserMenuProps={{
      menuTitle: 'Added menu',
      menuItems: [
        { title: 'Agent1', onClick: console.log },
        { title: 'Agent2',  href: 'http://www.example.com' },
      ],
      actionItem: {
        title: 'See more on this link',
        href: 'http://www.example.com',
        smallSize: true
      },
    }}
    tabNavigationProps={{
      tabs: [
        {
          title: 'Hello',
        },
        {
          title: 'Tabs',
        },
      ],
      selectedTabIndex: 0,
      onTabClick: console.log,
    }}
  />
`;

const withAllComponentsCode = `<Header
  title="My app title"
  userMenuProps={{
    menuTitle: 'User name',
    image:
      'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
    menuItems: [
      { title: 'Hello', onClick: console.log },
      { title: 'World', href: 'http://www.example.com' },
    ],
    actionItem: { title: 'Logout' },
  }}
  additionalUserMenuProps={{
    menuTitle: 'Added menu',
    menuItems: [
      {
        image: 'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
        title: 'Agent1',
        onClick: console.log
      },
      {
        image:
        'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
        title: 'Agent2',
        href: 'http://www.example.com'
      },
    ],
    actionItem: {
        title: 'See more on this link',
        href: 'http://www.example.com',
        smallSize: true
      },
  }}
  tabNavigationProps={{
    tabs: [
      {
        title: 'Hello',
      },
      {
        title: 'Tabs',
      },
    ],
    selectedTabIndex: 0,
    onTabClick: console.log,
  }}
/>
`;

const withTitleAndUserCode = `<Header
  userMenuProps={{
    image:
      'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
    menuItems: [
      { title: 'Hello', onClick: console.log },
      { title: 'World', href: 'http://www.example.com' },
    ],
    actionItem: { title: 'Logout' },
  }}

  tabNavigationProps={{
    tabs: [
      {
        title: 'Hello',
      },
      {
        title: 'Tabs',
      },
    ],
    selectedTabIndex: 0,
    onTabClick: console.log,
  }}
/>
`;

const withTabNavigationCode = `<Header
  userMenuProps={{
    image:
      'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
    menuItems: [
      { title: 'Hello', onClick: console.log },
      { title: 'World', href: 'http://www.example.com' },
    ],
    actionItem: { title: 'Logout' },
  }}

  tabNavigationProps={{
    tabs: [
      {
        title: 'Hello',
      },
      {
        title: 'Tabs',
      },
    ],
    selectedTabIndex: 0,
    onTabClick: console.log,
  }}
/>
`;
const Header: ComponentData = {
  description: `
    The Header represents a versatile component which can house a title or a TabNavigation, as well as a UserMenu and Additional User Menu.
  `,
  path: 'src/components/Header/index.js',
  usage,
  examples: [
    {
      title: 'With all 4 components',
      code: withAllComponentsCode,
    },
    {
      title: 'With title and user menu',
      code: withTitleAndUserCode,
    },
    {
      title: 'With tab navigation and user menu',
      code: withTabNavigationCode,
    }
  ],
  propsAndMethods: [
    { name: 'additionalUserMenuProps', type: 'UserMenuProps', default: 'null' },
    { name: 'className', type: 'string' },
    { name: 'style', type: 'Object', default: '{ }' },
    { name: 'tabNavigationProps', type: 'TabNavigationProps', default: 'null' },
    { name: 'title', type: 'string', default: 'null' },
    { name: 'userMenuProps', type: 'UserMenuProps', default: 'null' },
  ],
}

export default Header;