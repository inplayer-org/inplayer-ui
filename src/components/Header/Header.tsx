import React from 'react';
import styled from 'styled-components';
import { prop } from 'styled-tools';
import colors from 'theme/colors';

// Components
import UserMenu from '../UserMenu';
import { Props as UserMenuProps } from '../UserMenu/UserMenu';
import TabNavigation from '../TabNavigation';
import { Props as TabNavigationProps } from '../TabNavigation/TabNavigation';

const Container = styled.header`
  align-items: center;
  padding: 0 2%;
  background: ${colors.white};
  border-bottom: 1px solid ${colors.gray};
  display: flex;
  height: ${prop('height', 50)}px;
  justify-content: space-between;
  margin: 0;
  width: 100%;
  z-index: 3;
`;

const Title = styled.h3`
  color: ${({ theme }) => theme.palette.text.main};
  font-weight: ${({ theme }) => theme.font.weights.light};
  margin: 0;
`;

const MenuWrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

type Props = {
  /**
   * Header title
   */
  title?: string | null;
  /**
   * User menu item properties: `image`, `menuTitle`, `menuItems`, `actionItem` and `onClick`
   */
  userMenuProps?: UserMenuProps | null;
  /**
   * Additional user menu
   */
  additionalUserMenuProps?: UserMenuProps | null;
  /**
   * Tab related properties: `tabs`, `onTabClick`, `selectedTabIndex`
   */
  tabNavigationProps?: TabNavigationProps | null;
};

const Header = ({
  title = null,
  userMenuProps = null,
  additionalUserMenuProps = null,
  tabNavigationProps = null,
}: Props) => (
  <Container>
    {title && <Title>{title}</Title>}
    {tabNavigationProps && <TabNavigation {...tabNavigationProps} />}
    <MenuWrapper>
      {additionalUserMenuProps && <UserMenu {...additionalUserMenuProps} />}
      {userMenuProps && <UserMenu {...userMenuProps} />}
    </MenuWrapper>
  </Container>
);

export default Header;
