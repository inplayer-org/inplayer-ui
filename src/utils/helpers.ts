import moment from 'moment';

export const getMonthOptions = (): Array<any> =>
  moment.months().map((label, value) => ({ value, displayName: label }));

export const getYearOptions = () => {
  const yearOptions: Array<any> = [];
  for (let i = moment().year() - 100; i <= moment().year(); i += 1) {
    yearOptions.push({ value: i.toString(), displayName: i.toString() });
  }

  return yearOptions;
};
