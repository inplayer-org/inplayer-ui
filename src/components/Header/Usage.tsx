import React, { useState } from 'react';
import Header from '.';

const App = () => {
  const [tabSelected, setTabSelected] = useState(0);
  return (
    <Header
      title="My app title"
      userMenuProps={{
        menuTitle: 'User name',
        image: 'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
        menuItems: [
          { title: 'Hello', onClick: console.log },
          { title: 'World', href: 'http://www.example.com' },
        ],
        actionItem: { title: 'Logout' },
      }}
      additionalUserMenuProps={{
        menuTitle: 'Added menu',
        image:
          'https://media.sproutsocial.com/uploads/2017/02/10x-featured-social-media-image-size.png',
        menuItems: [
          { title: 'Agent1', onClick: console.log },
          { title: 'Agent2', href: 'http://www.example.com' },
        ],
        actionItem: {
          title: 'See more on this link',
          href: 'http://www.example.com',
          smallSize: true,
        },
      }}
      tabNavigationProps={{
        tabs: [
          {
            title: 'Hello',
          },
          {
            title: 'Tabs',
          },
        ],
        selectedTabIndex: tabSelected,
        onTabClick: (index: number) => setTabSelected(index),
      }}
    />
  );
};

export default App;
