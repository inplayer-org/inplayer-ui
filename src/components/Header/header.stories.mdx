import { useState } from 'react';
import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import Header from './'

<Meta title="Header" component={<Header/>} />

# Header

The Header represents a versatile component which can house a title or a TabNavigation, 
as well as a UserMenu and Additional User Menu.

```jsx
import { Header } from '@inplayer-org/inplayer-ui';

const Page = () => (
  <Header />
)
```
### Props
<Props of={Header}/>

### Header example
<Preview withSource="open">
  <Story name="demo">
    {() => {
        const [tabSelected, setTabSelected] = useState(0);
        return (
            <div style={{height: '250px', width: '90%'}}>
                <Header
                title="My app title"
                userMenuProps={{
                    menuTitle: 'User name',
                    image: 'https://www.impaact4tb.org/wp-content/uploads/2018/07/Generic-male-icon.png',
                    menuItems: [
                    { title: 'Hello', onClick: console.log },
                    { title: 'World', href: 'http://www.example.com' },
                    ],
                    actionItem: { title: 'Logout' },
                }}
                additionalUserMenuProps={{
                    menuTitle: 'Added menu',
                    image:
                    'https://media.sproutsocial.com/uploads/2017/02/10x-featured-social-media-image-size.png',
                    menuItems: [
                    { title: 'Agent1', onClick: console.log },
                    { title: 'Agent2', href: 'http://www.example.com' },
                    ],
                    actionItem: {
                    title: 'See more on this link',
                    href: 'http://www.example.com',
                    smallSize: true,
                    },
                }}
                tabNavigationProps={{
                    tabs: [
                    {
                        title: 'Hello',
                    },
                    {
                        title: 'Tabs',
                    },
                    ],
                    selectedTabIndex: tabSelected,
                    onTabClick: (index) => setTabSelected(index),
                }}
                />
            </div>
        );
    }}
  </Story>
</Preview>

