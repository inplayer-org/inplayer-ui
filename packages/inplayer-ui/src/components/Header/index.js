// @flow
import React from 'react';
import styled from 'styled-components';
import { prop } from 'styled-tools';
import colors from 'config/colors';
import { uiColors, fontWeights } from 'utils';

// Components
import UserMenu, { type UserMenuProps } from 'components/UserMenu';
import TabNavigation, { type TabNavigationProps } from 'components/TabNavigation';

const Container = styled.header`
  align-items: center;
  padding: 0 2%;
  background: ${colors.white};
  border-bottom: 1px solid ${colors.gray};
  display: flex;
  height: ${prop('height', 50)}px;
  justify-content: space-between;
  margin: 0;
  width: 100%;
  z-index: 3;
`;

const Title = styled.h3`
  color: ${uiColors('text.main')};
  font-weight: ${fontWeights('light')};
  margin: 0;
`;

const MenuWrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

type HeaderProps = {
  title?: string,
  userMenuProps?: UserMenuProps,
  additionalUserMenuProps?: UserMenuProps,
  tabNavigationProps?: TabNavigationProps,
  className?: string,
  style?: Object,
};

const Header = ({
  title,
  userMenuProps,
  additionalUserMenuProps,
  tabNavigationProps,
  className,
  style,
}: HeaderProps) => (
  <Container className={className} style={style}>
    {title && <Title>{title}</Title>}
    {tabNavigationProps && <TabNavigation {...tabNavigationProps} />}
    <MenuWrapper>
      {additionalUserMenuProps && <UserMenu {...additionalUserMenuProps} />}
      {userMenuProps && <UserMenu {...userMenuProps} />}
    </MenuWrapper>
  </Container>
);

Header.defaultProps = {
  title: null,
  userMenuProps: null,
  additionalUserMenuProps: null,
  tabNavigationProps: null,
  className: '',
  style: {},
};

export default Header;
