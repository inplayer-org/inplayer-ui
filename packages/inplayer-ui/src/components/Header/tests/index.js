import { generateSnapshots } from 'packages/inplayer-ui/src/components/Header/tests/helpers';

import Header from '../index';

generateSnapshots(Header);
