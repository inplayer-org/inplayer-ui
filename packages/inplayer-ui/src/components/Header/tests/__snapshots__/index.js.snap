// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 2%;
  background: #fff;
  border-bottom: 1px solid #d7dde5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 50px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0;
  width: 100%;
  z-index: 3;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<header
  className="c0"
  style={Object {}}
>
  <div
    className="c1"
  />
</header>
`;
